src/server.ts

```diff
@@ -70,36 +62,42 @@ function toUsername (name: string) {
 github.on('push', function push (repo, ref, result) {
   var url = result.compare
   var branch = /[^/]+$/.exec(ref)[0]
-  shorten(url, function pushShortened (url) {
-    if (branch !== 'master') return
-    var messages: string[] = []
-    var staffMessages: string[] = []
-
-    result.commits.forEach(function (commit) {
-      var commitMessage = commit.message
-      var shortCommit = /.+/.exec(commitMessage)[0]
-      if (commitMessage !== shortCommit) {
-        shortCommit += '…'
-      }
-      // result.sender.login here is the login of user which performed the push,
-      // not the original author of the commit. We don't have the GitHub login for
-      // the user, the best we have for attribution is the commit's author's name.
-      var username = toUsername(commit.author.name)
-      const repoName = getRepoName(repo)
-      const { url } = commit
-      const id = commit.id.substring(0, 6)
-      const formattedRepo = `[<font color='FF00FF'>${h(repoName)}</font>]`
-      const formattedUserName = `<font color='909090'>(${h(username)})</font>`
-      messages.push(`${formattedRepo} <a href=\"${h(url)}\"><font color='606060'>${h(id)}</font></a> ${h(shortCommit)} ${formattedUserName}`)
-      staffMessages.push(`${formattedRepo} <a href=\"${h(url)}\">${h(shortCommit)}</a> ${formattedUserName}`)
-    })
-    client.report('/addhtmlbox ' + messages.join('<br>'))
-    if (reposToReportInStaff.has(repo)) {
-      client.reportStaff('/addhtmlbox ' + staffMessages.join('<br>'))
-    }
+  if (branch !== 'master') return
+  var promises = []
+  var messagesPS: string[] = []
+  var messagesDiscord: string[] = []
+  var staffMessages: string[] = []
+
+  result.commits.forEach(function (commit) {
+    promises.push(new Promise(function (resolve, reject) {
```

Comment:
This looks like `.map` for me.

Reply:
Do you mean instead of ``.push``? Isn't ``.map`` usually used to manipulate existing elements of an array?

Reply:
In Node.js, you can transform an array using a function.

```javascript
> [1, 2, 3].map(x => x + 1)
[ 2, 3, 4 ]
```

This way, you can avoid using `promises.push`, and creating a `promises` array.

Reply:
You mean to do something like ``commit.map(funtion() { commit.url = await shorten(url, ...``?

Reply:
```javascript
result.commits.map(commit => new Promise(resolve => {
  shorten(commit.url, url => {
    ...
    resolve()
  })
}))
```